lib/PipelineC/PipelineC.cpp

```diff
@@ -725,41 +722,42 @@ inline bool rp_manager_apply_diff_impl(rp_manager *manager,
   revng_check(diff != nullptr);
   revng_check(global_name != nullptr);
 
+  ExistingOrNew<rp_error> Error(error);
   auto &GlobalsMap = manager->context().getGlobals();
   auto Buffer = llvm::MemoryBuffer::getMemBuffer(diff);
 
   auto GlobalOrError = GlobalsMap.get(global_name);
   if (!GlobalOrError) {
-    llvmErrorToRpError(GlobalOrError.takeError(), error);
+    llvmErrorToRpError(GlobalOrError.takeError(), Error);
     return false;
   }
 
   auto &Global = GlobalOrError.get();
   auto MaybeDiff = Global->deserializeDiff(*Buffer);
   if (!MaybeDiff) {
-    llvmErrorToRpError(MaybeDiff.takeError(), error);
+    llvmErrorToRpError(MaybeDiff.takeError(), Error);
     return false;
   }
 
   auto &Diff = MaybeDiff.get();
   auto GlobalClone = Global->clone();
-  auto Error = GlobalClone->applyDiff(Diff);
-  if (Error) {
-    llvmErrorToRpError(std::move(Error), error);
+  auto ApplyError = GlobalClone->applyDiff(Diff);
+  if (ApplyError) {
+    llvmErrorToRpError(std::move(ApplyError), Error);
     return false;
   }
 
-  if (auto Error = GlobalClone->verify(); Error) {
-    **error = rp_simple_error(std::string("could not verify ") + global_name,
-                              "");
+  if (!GlobalClone->verify()) {
```

Comment:
Everywhere. More readable.

```suggestion
  if (not GlobalClone->verify()) {
```

Reply:
Done